### ISSUE TITLE
nextest support was recently broken, now attempts to exec a JSON file

### ISSUE THREAD

**Issue by saethlin** [rust-lang/miri#2437]
If I create an otherwise-empty Cargo project with this in `src/main.rs`:
```rust
fn main() {}

#[test]
fn foo() {}
```
Then `cargo +miri miri nextest list` (or `run`) now emits this:
```
error: creating test list failed

Caused by:
  for `scratch::bin/scratch`, running command `/tmp/scratch/target/miri/x86_64-unknown-linux-gnu/debug/deps/scratch-8097932b18eafdc6 --list --format terse` failed to execute

Caused by:
  Permission denied (os error 13)
```
As far as I can tell from running `strace -f`, there's an `execve` of a file under `target/x86_64-unknown-linux-gnu/debug/deps/` which fails with EACCESS, because the file is just some JSON and not marked as executable.

**Comment by RalfJung** [comment]
I suspect this is due to https://github.com/rust-lang/miri/pull/2428. Previously we used env vars to configure the target runner for cargo, now we use `--config` flag. My guess is that nextest does not properly propagate these `--config` flags when it recursively invokes cargo.

**Comment by saethlin** [comment]
Yup, did a little experimentation to confirm that this is a nextest issue: https://github.com/nextest-rs/nextest/issues/400

**Comment by sunshowers** [comment]
(side note: it would be really cool if [`cargo config`](https://doc.rust-lang.org/cargo/reference/unstable.html#cargo-config) were stable, it would avoid this class of issues =) )

**Comment by RalfJung** [comment]
FWIW, Miri is happy to use nightly-only features.

Or do you mean that nextest needs to *query* the config and would like to use `cargo config` for that?

**Comment by sunshowers** [comment]
The latter, yeah. Nextest needs to figure out what the config is and currently it emulates cargo's config search and discovery for that. If we could just ask cargo to do that for us, life would be much simpler.

**Comment by sunshowers** [comment]
This is addressed in cargo-nextest 0.9.29 -- binaries should be out within the next 20 minutes.
